### PR TITLE
JWT Authentication

### PR DESCRIPTION
**Pull request by AY-Chen**
WIP server authentication with JWT. Added new routes to deal with login and added a token requirement to the functions relating to controlling the blinds(ie scheduling and commands). 

Source code adapted from this tutorial: https://www.youtube.com/watch?v=WxGBoY5iNXY

A few more adjustments are needed, but at this point the auth is technically "working".

### CODE REVIEW

File: piserver/app.py [hunk 1]
```diff
@@ -50,7 +64,52 @@
...
+class User( db.Model ):
+    id = db.Column( db.Integer, primary_key=True )
+    public_id = db.Column( db.String(50), unique=True )
+    name = db.Column( db.String(30), unique=True )
+    password = db.Column( db.String(50) )
+
+'''
+Decorator for using the JWT token
+'''
+def token_required( fxn ):
```
Comment (ishchow) [inline piserver/app.py]: Are we able to check the ip the request comes from and if it's coming from the same machine as the server, just ignore auth?

Comment (AY-Chen) [reply piserver/app.py]: i think it's hard when the host is set to like 0.0.0.0 or something like that. I also thought that we'd sometimes want to test the auth locally anyway. I can look into it if you'd like.

**Comment by ishchow** [comment]
@AY-Chen also, how significantly did you use the example from that linked tutorial? Would it be wise to add an attribution to the code somewhere, maybe at the top of the file?

**Comment by AY-Chen** [comment]
> 
> 
> @AY-Chen also, how significantly did you use the example from that linked tutorial? Would it be wise to add an attribution to the code somewhere, maybe at the top of the file?

Pretty significantly. I was gonna put it in the report but yeah, in the file would be a good idea too.

**Comment by AY-Chen** [comment]
> 
> 
> Looks good! LGTM!

Thanks Ishaat :) I'm going to make a few more small changes to it to clean it up before i merge it.

**Pull request merged**